tests/www/views/conftest.py

```diff
@@ -0,0 +1,190 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from contextlib import contextmanager
+from typing import Any, Dict, Generator, List, NamedTuple
+
+import flask
+import jinja2
+import pytest
+
+from airflow import settings
+from airflow.models import DagBag
+from airflow.www.app import create_app
+from tests.test_utils.api_connexion_utils import create_user, delete_roles
+from tests.test_utils.decorators import dont_initialize_flask_app_submodules
+from tests.test_utils.www import client_with_login
+
+
+@pytest.fixture(autouse=True, scope="module")
+def session():
+    settings.configure_orm()
+    yield settings.Session
```

Comment:
Shouldn't this be called?

```suggestion
    yield settings.Session()
```

Reply:
No, this is intentionally returning a shared session object (`settings.Session` is a `sqlalchemy.orm.session.Session` *instance* -- named pretty poorly IMO but well). If we want a new session object we can just use `create_session()` instead (which is done extensively).

This shared session is only used in very selected places. I think some of them can actually be replaced by `create_session()` (and did replace somein this PR), but didn’t bother to check all of them TBH.

Reply:
Create session does `airflow.settings.Session()` -- accessing properties of the scoped_session object is actually proxying to the session object

https://github.com/sqlalchemy/sqlalchemy/blob/adb3a302ef1f25ae41b2a83ce857e3ed2613c84d/lib/sqlalchemy/orm/scoping.py#L67

Either way the behaviour is the same, it just feels more "correct" to me to get the actual session object back, not the magic proxy.